packages/utils/src/dataURItoBlob.ts

```diff
@@ -4,39 +4,36 @@
  * @param dataURI - The `DataUrl` potentially containing name and raw data to be converted to a Blob
  * @returns - an object containing a Blob and its name, extracted from the URI
  */
-export default function dataURItoBlob(dataURI: string) {
-  // Split metadata from data
-  const splitted: string[] = dataURI.split(',');
-  // Split params
-  const params: string[] = splitted[0].split(';');
-  // Get mime-type from params
-  const type: string = params[0].replace('data:', '');
-  // Filter the name property from params
-  const properties = params.filter((param) => {
-    return param.split('=')[0] === 'name';
-  });
-  // Look for the name and use unknown if no name property.
-  let name: string;
-  if (properties.length !== 1) {
-    name = 'unknown';
-  } else {
-    // Because we filtered out the other property,
-    // we only have the name case here, which we decode to make it human-readable
-    name = decodeURI(properties[0].split('=')[1]);
+export default function dataURItoBlob(dataURILike: string) {
+  // check if is dataURI
+  if (dataURILike.indexOf('data:') === -1) {
+    throw new Error('File is invalid: URI must be a dataURI');
   }
+  const dataURI = dataURILike.slice(5);
+  const splitted = dataURI.split(';base64,');
+  // check if the dataURI is base64
+  if (splitted.length !== 2) {
+    throw new Error('File is invalid: dataURI must be base64');
+  }
+  const [media, base64] = splitted;
+  const [mime, ...mediaparams] = media.split(';');
+  const type = mime || '';
+  const name = decodeURI(
+    mediaparams.map((param) => param.split('=')).find(([key]) => key === 'name')?.[1] || 'unknown'
+  );
 
   // Built the Uint8Array Blob parameter from the base64 string.
   try {
-    const binary = atob(splitted[1]);
-    const array = [];
+    const binary = atob(base64);
+    const array = new Array(binary.length);
     for (let i = 0; i < binary.length; i++) {
-      array.push(binary.charCodeAt(i));
+      array[i] = binary.charCodeAt(i);
     }
     // Create the blob object
     const blob = new window.Blob([new Uint8Array(array)], { type });
 
     return { blob, name };
   } catch (error) {
-    return { blob: { size: 0, type: (error as Error).message }, name: dataURI };
+    throw new Error('File is invalid: failed to decode base64');
```

Comment:
Why not just treat all of the errors to return a blob like these?

Reply:
The original code treated the invalid file as zero-sized file. Of course, the invalid file is different from the zero-sized valid file. Invalid files should have been taken care of by the callee, but it is not.
This behavior also allowed `javascript:` schema.